File: node/src/wallet.ts
/**
 * Wallet service for the Multando API.
 */

import { HttpClient } from './http.js';
import type { Activity, PaginatedResponse, WalletBalance } from './models.js';

export class WalletService {
  constructor(private readonly http: HttpClient) {}

  async balance(): Promise<WalletBalance> {
    return this.http.get<WalletBalance>('/wallet/info');
  }

  async activities(page?: number): Promise<PaginatedResponse<Activity>> {
    const params: Record<string, string> = {};
    if (page !== undefined) params.page = String(page);
    return this.http.get<PaginatedResponse<Activity>>('/wallet/activities', params);
  }
}

File: python/multando/wallet.py
"""Wallet service."""

from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

from .models import ActivityList, WalletBalance

if TYPE_CHECKING:
    from ._http import HttpClient, SyncHttpClient


class WalletService:
    """Async wallet operations."""

    def __init__(self, http: "HttpClient") -> None:
        self._http = http

    async def balance(self) -> WalletBalance:
        data = await self._http.get("/wallet/info")
        return WalletBalance.from_dict(data)

    async def activities(self, page: int = 1) -> ActivityList:
        params: Dict[str, Any] = {"page": page}
        data = await self._http.get("/wallet/withdrawals", params=params)
        return ActivityList.from_dict(data)


class SyncWalletService:
    """Synchronous wallet operations."""

    def __init__(self, http: "SyncHttpClient") -> None:
        self._http = http

    def balance(self) -> WalletBalance:
        data = self._http.get("/wallet/info")
        return WalletBalance.from_dict(data)

    def activities(self, page: int = 1) -> ActivityList:
        params: Dict[str, Any] = {"page": page}
        data = self._http.get("/wallet/withdrawals", params=params)
        return ActivityList.from_dict(data)
